terraform/github/components.tf

```diff
@@ -0,0 +1,109 @@
+module "config_component" {
+  source = "./modules/component"
+
+  ci_team     = github_team.ci
+  name        = "config"
+  description = "Common library for configuring Athena components"
+  url         = "https://athenaframework.org/Config/"
+  topics      = ["crystal"]
+}
+
+module "console_component" {
+  source = "./modules/component"
+
+  ci_team     = github_team.ci
+  name        = "console"
+  description = "Allows for the creation of CLI based commands"
+  url         = "https://athenaframework.org/Console/"
+  topics      = ["crystal"]
+}
+
+module "dependency-injection_component" {
+  source = "./modules/component"
+
+  ci_team     = github_team.ci
```

Comment:
Not sure if there's a better way to handle this?

Reply:
"Better" is debatable, the decisions of Hashicorp to not allow global variables between parent states and their modules was explicit as they prefer a clean contract between the inputs/outputs. 

With all of that said, some _alternative_ solutions if copy/pasting that same value really gets under your skin.

1. Push the CI team value to some secrets manager/param manager/etc and leverage a `data` source within the module to not need to explicitly pass it in. In AWS for example, you could publish the output info into [Parameter Store](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html) to a path (say, `/github/team`  that contains the json for the output values) and and have the module itself read the value from the paramstore path. Then the dependency is _very implicit_ where the module will fail to grab the data value should it not exist. 

2. Use another tool to template terraform itself, e.g. terragrunt. This could help out in the "anti-DRY with the provider configuration" as well. My team hasn't adopted it yet, but have considered it. See: https://terragrunt.gruntwork.io/docs/features/keep-your-terraform-code-dry/#dry-common-terraform-code-with-terragrunt-generate-blocks